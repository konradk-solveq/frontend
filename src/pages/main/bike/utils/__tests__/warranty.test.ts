import {
    getWarrantyStatusInfo,
    getWarrantyStatusInfoText,
    getWarrantyTypeText,
    shouldShowWarrantyStatusInfo,
} from '@pages/main/bike/utils/warranty';

describe('warranty - pages/main/bikeutils', () => {
    describe('getWarrantyStatusInfo()', () => {
        it('should return warranty status info', () => {
            const result = getWarrantyStatusInfo(true);

            expect(result).toMatchSnapshot();
        });

        it('should return contact info', () => {
            const result = getWarrantyStatusInfo(false);

            expect(result).toMatchSnapshot();
        });

        afterEach(() => {
            jest.restoreAllMocks();
            jest.clearAllMocks();
        });
    });

    describe('shouldShowWarrantyStatusInfo()', () => {
        it('should return true when warranty type is "extended"', () => {
            const result = shouldShowWarrantyStatusInfo('extended');

            expect(result).toBe(true);
        });

        it('should return true when warranty type is "extended" and "daysToEnd" is smaller than "0"', () => {
            const result = shouldShowWarrantyStatusInfo('extended');

            expect(result).toBe(true);
        });

        it('should return true when warranty type is "no-info"', () => {
            const result = shouldShowWarrantyStatusInfo('no-info');

            expect(result).toBe(true);
        });

        it('should return true when warranty type is "no-info" and "daysToEnd" is smaller than "0"', () => {
            const result = shouldShowWarrantyStatusInfo('no-info');

            expect(result).toBe(true);
        });

        it('should return true when warranty daysToEnd is bigger than "0"', () => {
            const result = shouldShowWarrantyStatusInfo('basic', 2);

            expect(result).toBe(true);
        });

        it('should return false when warranty daysToEnd is lower than "0"', () => {
            const result = shouldShowWarrantyStatusInfo('basic', -2);

            expect(result).toBe(false);
        });

        it('should return false when warranty daysToEnd is equal to "0"', () => {
            const result = shouldShowWarrantyStatusInfo('basic', 0);

            expect(result).toBe(false);
        });

        afterEach(() => {
            jest.restoreAllMocks();
            jest.clearAllMocks();
        });
    });

    describe('getWarrantyTypeText()', () => {
        it('should return "no info" warranty type text when "caontainsInfo" is equal to "false"', () => {
            const result = getWarrantyTypeText(true, 'warranty info', false);

            expect(result).toMatchSnapshot();
        });

        it('should return warranty type text when "caontainsInfo" is equal to "true"', () => {
            const result = getWarrantyTypeText(true, 'warranty info', true);

            expect(result).toMatchSnapshot();
        });

        it('should return "expired" warranty type text when "caontainsInfo" is equal to "true" and "withInfo" is equal to "false"', () => {
            const result = getWarrantyTypeText(false, 'warranty info', true);

            expect(result).toMatchSnapshot();
        });

        afterEach(() => {
            jest.restoreAllMocks();
            jest.clearAllMocks();
        });
    });

    describe('getWarrantyStatusInfoText()', () => {
        it('should return contact info text when "type" is euqal to "no-info"', () => {
            const result = getWarrantyStatusInfoText('no-info');

            expect(result).toMatchSnapshot();
        });

        it('should return "lietime" status info text when "type" is euqal to "extended"', () => {
            const result = getWarrantyStatusInfoText('extended');

            expect(result).toMatchSnapshot();
        });

        it('should return "daysToEnd" status info text when "type" is euqal to "basic"', () => {
            const result = getWarrantyStatusInfoText('basic', 2);

            expect(result).toMatchSnapshot();
        });

        afterEach(() => {
            jest.restoreAllMocks();
            jest.clearAllMocks();
        });
    });
});
