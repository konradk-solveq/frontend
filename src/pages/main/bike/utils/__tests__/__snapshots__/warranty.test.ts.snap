// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`warranty - pages/main/bikeutils getWarrantyStatusInfo() should return contact info 1`] = `"MainBike.warranty.missingToEnd"`;

exports[`warranty - pages/main/bikeutils getWarrantyStatusInfo() should return warranty status info 1`] = `"MainBike.warranty.toEnd"`;

exports[`warranty - pages/main/bikeutils getWarrantyStatusInfoText() should return "daysToEnd" status info text when "type" is euqal to "basic" 1`] = `"2 MainBike.warranty.days"`;

exports[`warranty - pages/main/bikeutils getWarrantyStatusInfoText() should return "lietime" status info text when "type" is euqal to "extended" 1`] = `"MainBike.warranty.lifetime"`;

exports[`warranty - pages/main/bikeutils getWarrantyStatusInfoText() should return contact info text when "type" is euqal to "no-info" 1`] = `"MainBike.warranty.missingToEndAction"`;

exports[`warranty - pages/main/bikeutils getWarrantyTypeText() should return "expired" warranty type text when "caontainsInfo" is equal to "true" and "withInfo" is equal to "false" 1`] = `"MainBike.warranty.expired"`;

exports[`warranty - pages/main/bikeutils getWarrantyTypeText() should return "no info" warranty type text when "caontainsInfo" is equal to "false" 1`] = `"MainBike.warranty.noInfo"`;

exports[`warranty - pages/main/bikeutils getWarrantyTypeText() should return warranty type text when "caontainsInfo" is equal to "true" 1`] = `"warranty info"`;
